static/game/powerups/machinegun.js

```diff
@@ -0,0 +1,53 @@
+import { PowerUp } from "../powerup.js";
+import { Vector } from "../../modules/vector.js";
+import { Creature } from "../creature.js";
+
+const MIN_FIRE_DELAY = 2;
+const FIRE_DELAY_FACTOR = 1;
```

Comment:
I'm a little worried about a fire delay factor since fire delay should only be an int since it's in units of game steps. I guess we could still double how effective the power-up is.

Reply:
Just don't set FIRE_DELAY_FACTOR lower than 1